src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -549,9 +587,27 @@ export class ReplaceAction extends AbstractSearchAndReplaceAction {
 
 	override async run(): Promise<any> {
 		this.enabled = false;
+		let elementsToReplace: RenderableMatch[] = [];
+		const currentSelection: RenderableMatch[] = [];
+
+		this.viewer.getSelection().forEach((elem) => {
+			currentSelection.push(<RenderableMatch>elem);
+		});
+
+		// if you're using multiselect and you click to replace one element from the tree, replace all selected
+		if (currentSelection.length > 1 && currentSelection.includes(this.element)) {
```

Comment:
Maybe worth it to have a helper function for all the actions to do this common work of looking at the multiselection and figuring out what element you are operating on?